pwnlib/fmtstr.py

```diff
@@ -102,7 +108,497 @@ def send_payload(payload):
+
+    Examples:
+        >>> initial = {(0,0): (0,0,0), (0,1): None, (1,0): None, (1,1): None}
+        >>> pwnlib.fmtstr.find_min_hamming_in_range_step(initial, (0, 0xFF, 0x1), 0, 0)
+        (1, 1, 255)
+        >>> pwnlib.fmtstr.find_min_hamming_in_range_step(initial, (0, 1, 1), 0, 0)
+        (1, 1, 255)
+        >>> pwnlib.fmtstr.find_min_hamming_in_range_step(initial, (0, 1, 1), 0, 1)
+        (0, 0, 0)
+        >>> pwnlib.fmtstr.find_min_hamming_in_range_step(initial, (0, 1, 0), 0, 1)
+        (1, 0, 255)
+        >>> repr(pwnlib.fmtstr.find_min_hamming_in_range_step(initial, (0xFF, 0x00, 0xFF), 1, 0))
+        'None'
+    """
+    lower, upper, value = step
+    carryadd = 1 if carry else 0
+
+    valbyte = value & 0xFF
+    lowbyte = lower & 0xFF
+    upbyte = upper & 0xFF
+
+    # if we can the requested byte without carry, do so
+    # requiring strictness if possible is not a problem since strictness will cost at most a single byte
+    # (so if we don't get our wanted byte without strictness, we may as well require it if possible)
+    val_require_strict = valbyte > upbyte or valbyte == upbyte and strict
+    if lowbyte + carryadd <= valbyte:
+        if prev[(0, val_require_strict)]:
+            prev_score, prev_val, prev_mask = prev[(0, val_require_strict)]
+            return prev_score + 1, (prev_val << 8) | valbyte, (prev_mask << 8) | 0xFF
+
+    # now, we have two options: pick the wanted byte (forcing carry), or pick something else
+    # check which option is better
+    lowcarrybyte = (lowbyte + carryadd) & 0xFF
+    other_require_strict = lowcarrybyte > upbyte or lowcarrybyte == upbyte and strict
+    other_require_carry = lowbyte + carryadd > 0xFF
+    prev_for_val = prev[(1, val_require_strict)]
+    prev_for_other = prev[(other_require_carry, other_require_strict)]
+    if prev_for_val and (not prev_for_other or prev_for_other[0] <= prev_for_val[0] + 1):
+        return prev_for_val[0] + 1, (prev_for_val[1] << 8) | valbyte, (prev_for_val[2] << 8) | 0xFF
+    if prev_for_other:
+        return prev_for_other[0], (prev_for_other[1] << 8) | lowcarrybyte, (prev_for_other[2] << 8)
+    return None
+
+def find_min_hamming_in_range(maxbytes, lower, upper, target):
+    """
+    Find the value which differs in the least amount of bytes from the target and is in the given range.
+
+    Returns a tuple (count, value, mask) where count is the number of equal bytes and mask selects the equal bytes.
+    So mask & target == value & target and lower <= value <= upper.
+
+    Arguments:
+        maxbytes(int): bytes above maxbytes (counting from the least significant first) don't need to match
+        lower(int): lower bound for the returned value, inclusive
+        upper(int): upper bound, inclusive
+        target(int): the target value that should be approximated
+
+    Examples:
+        >>> pp = lambda (score, value, mask): (score, hex(value), hex(mask))
```

Comment:
Tuple parameters were [removed](https://www.python.org/dev/peps/pep-3113/) in python3 (#1119).

This regression could be fixed for example like this:
```suggestion
        >>> pp = lambda svm: (svm[0], hex(svm[1]), hex(svm[2]))
```